app/components/download-switch/download-switch.tsx

```diff
@@ -224,17 +228,21 @@ export function DownloadSwitch({
               case DownloadStatus.Downloading: {
                 if (downloadStatusToBe === DownloadStatusToBe.NotDownloaded) {
                   __DEV__ && console.log(`Pausing download of track ${track.trackId}`)
-                  await resumableDownloads.get(track.trackId).pauseAsync()
-                  statuses.set(track.trackId, DownloadStatus.Paused)
+                  const download = resumableDownloads.get(track.trackId)
```

Comment:
I don't like the fact that it is possible at all for `download` to be `undefined` but I understand why it's the case.

Reply:
Yeah, me neither. Maybe there's some type level magic possible which gives us some kind of proof earlier, that the track is really part of the collection. Similar to https://kataskeue.com/gdp.pdf.